### PR TITLE
part of scrutinizer fixes

### PR DESCRIPTION
**Pull request by EmanueleMinotto**
Ref: https://github.com/puli/issues/issues/92

### CODE REVIEW

File: src/AbstractPathMappingRepository.php [hunk 1]
```diff
@@ -208,7 +208,7 @@ protected function sortStore()
     /**
      * Create a filesystem or generic resource.
      *
-     * @param string $filesystemPath
+     * @param string|bool|null $filesystemPath
```
Comment (webmozart) [inline src/AbstractPathMappingRepository.php]: Why bool?

File: src/InMemoryRepository.php [hunk 2]
```diff
@@ -81,7 +81,7 @@ public function find($query, $language = 'glob')
         $resources = array();
 
         if (Glob::isDynamic($query)) {
-            $resources = $this->getGlobIterator($query);
+            $resources = iterator_to_array($this->getGlobIterator($query));
```
Comment (webmozart) [inline src/InMemoryRepository.php]: This isn't needed as the ArrayResourceCollection accepts iterators. You could add `Traversable` to the constructor argument's PHPDoc of that class instead.

File: src/InMemoryRepository.php [hunk 3]
```diff
@@ -183,7 +183,7 @@ public function listChildren($path)
     {
         $iterator = $this->getChildIterator($this->get($path));
 
-        return new ArrayResourceCollection($iterator);
+        return new ArrayResourceCollection(iterator_to_array($iterator));
```
Comment (webmozart) [inline src/InMemoryRepository.php]: same here

File: src/AbstractPathMappingRepository.php (hunk 1 above)
Comment (EmanueleMinotto) [inline src/AbstractPathMappingRepository.php]: Misunderstood a scrutinizer suggestion, fixed now

File: src/InMemoryRepository.php (hunk 2 above)
Comment (EmanueleMinotto) [inline src/InMemoryRepository.php]: ArrayResourceCollection currently requires both `Traversable` and `ArrayAccess` (afaik), but there isn't an interface extending both them (and the creation just to fill a PHPDoc seems useless to me), so I would suggest to revert this change and just ignore the Scrutinizer issue until next changes to the ArrayResourceCollection.

What to you think? Did I miss something?

Comment (webmozart) [inline src/InMemoryRepository.php]: No, ArrayResourceCollection just requires any `Traversable` object - you can see that that object is passed to `replace()`, which in turn calls `iterator_to_array()`.

File: src/Resource/Collection/ArrayResourceCollection.php [hunk 4]
```diff
@@ -40,7 +40,7 @@ class ArrayResourceCollection implements IteratorAggregate, ResourceCollection
      * You can pass the resources that you want to initially store in the
      * collection as argument.
      *
-     * @param PuliResource[] $resources The resources to store in the collection.
+     * @param PuliResource[]|\Traversable $resources The resources to store in the collection.
```
Comment (webmozart) [inline src/Resource/Collection/ArrayResourceCollection.php]: Can you change this to `Traversable` (intead of `\Traversable`) and import the class instead? Also please add this as well for `replace()` and `merge()` (in the `ResourceCollection` interface).
